### ISSUE TITLE
Yandex Lockbox Connection Leaks

### ISSUE THREAD

**Issue by ashikhman** [external-secrets/external-secrets#833]
**Description**
We ran into a connection leak bug when using Yandex Lockbox provider. Someday we deleted service account that is used to update the secret and since that day, the logs recorded an increase in connections.

On the graph below you can see increase in connections when we start External Secrets pod and then drop in connections on pod stop.
![Screenshot from 2022-03-16 11-44-43](https://user-images.githubusercontent.com/3223577/158551794-087f67c6-74e2-4178-82c3-50f8833977d8.png)

Also there is error log everytime External Secrets tries to update the secret data:
```
{
  "level": "error",
  "ts": 1647364910.23199,
  "logger": "controllers.ExternalSecret",
  "msg": "could not get provider client",
  "ExternalSecret": "gateway-java-test/gateway-java",
  "SecretStore": "gateway-java-test/gateway-java",
  "error": "failed to create Yandex Lockbox client: iam token create failed: rpc error: code = Unauthenticated desc = Key 'aje9guud23i28bo0aog6' was not found",
  "stacktrace": "sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:298\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:253\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:214"
}
```

**How to reproduce**
1. Setup working External Secrets environment by following the docs https://cloud.yandex.ru/docs/managed-kubernetes/tutorials/kubernetes-lockbox-secrets
2. Set `refreshInterval` to a small value like 2 secs.
3. Delete service account `eso-service-account`.
4. Check pod logs and connections.

**Cause of the leak**
It looks like the root cuase of the bug is in [grpc.CreateLockboxClient#38](https://github.com/external-secrets/external-secrets/blob/29d2418d8e0b71d968d53864f9f3dc4030d97b94/pkg/provider/yandex/lockbox/client/grpc/grpc.go#L48) Call to `sdk.ApiEndpoint().ApiEndpoint().Get` returns error `iam token create failed` and then immediately returns from the calling func without executing `CloseSDK`. Also there is similar behaviour in `CreateIamToken` func.
![Screenshot from 2022-03-15 22-07-11](https://user-images.githubusercontent.com/3223577/158556049-797b781b-c67f-4d3f-93d9-e258d3818a8a.png)

**Comment by AndreyZamyslov** [comment]
I'll try to reproduce and prepare a fix for that

**Comment by moolen** [comment]
fixed with #835
